subsys/audio_module/audio_module.c

```diff
@@ -699,43 +688,71 @@ int audio_module_configuration_get(struct audio_module_handle const *const handl
  *
  */
 int audio_module_connect(struct audio_module_handle *handle_from,
-			 struct audio_module_handle *handle_to)
+			 struct audio_module_handle *handle_to, bool connect_external)
 {
 	int ret;
 	struct audio_module_handle *handle;
 
-	if (handle_from == NULL || handle_to == NULL) {
+	if (handle_from == handle_to) {
+		LOG_WRN("Module handles identical");
+		return -EINVAL;
+	}
+
+	if (handle_from == NULL) {
 		LOG_ERR("Invalid parameter for the connection function");
 		return -EINVAL;
 	}
 
-	ret = handle_connection_type_test(handle_from, handle_to);
-	if (ret) {
-		LOG_ERR("Connections between these modules, %s to %s, is not supported",
-			handle_from->name, handle_to->name);
+	if (handle_from->description->type != AUDIO_MODULE_TYPE_INPUT &&
```

Comment:
It is the same here as the states: We should check for valid module types, and not the invalid ones in case we expand with a different kind of module later.